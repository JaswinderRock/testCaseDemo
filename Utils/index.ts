export const findByTestAttr = (component: any, attr: String) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}
export const findByClassName = (component: any, className: String) => {
    const wrapper = component.find(`.${className}`);
    return wrapper;
}