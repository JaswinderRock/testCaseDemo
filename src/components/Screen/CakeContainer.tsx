import React, { Component } from 'react';
import { connect } from 'react-redux';
import { buyCake } from '../Action/action';

type CakeProps = {
    numOfCakes: number;
    buyCake: () => void;
};
class CakeContainer extends Component<CakeProps>{
    render() {
        return (
            <div style={{ display: 'flex' }}>
                <div style={{ margin: 'auto' }}>
                    <h2>Number of Cakes {this.props.numOfCakes}</h2>
                    <button style={{ display: 'flex', margin: 'auto', borderRadius: '4px', textAlign: 'center', color: "white", background: 'blue', padding: '10px', fontSize: '24px', }} onClick={this.props.buyCake}>Buy Cake</button>
                </div>
            </div>
        )

    }
}
function mapStateToProps(state: any) {
    return { numOfCakes: state.numberOfCakes };
}
export default connect(mapStateToProps, { buyCake })(CakeContainer);