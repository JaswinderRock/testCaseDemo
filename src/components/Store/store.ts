import { createStore } from 'redux';
import { cakeReducer } from '../Reducer/cakeReducer';
import reducer from '../Reducer/incrementReducer';

export const incrementStore = createStore(reducer)
export const Store = createStore(cakeReducer);