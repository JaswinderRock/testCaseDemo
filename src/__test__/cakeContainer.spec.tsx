import React from 'react';
import { mount } from 'enzyme';
import CakeContainer from '../components/Screen/CakeContainer';
import { debug } from 'console';
import { Provider } from 'react-redux';
import { Store } from '../components/Store/store';
import { findByClassName, findByTestAttr } from '../../Utils';

const setUp = () => {
    const component = mount(<Provider store={Store}>
        <CakeContainer />
    </Provider>);
    return component;
}

describe("Render CakeContainer ", () => {
    let component: any;
    beforeEach(() => {
        component = setUp();
    });

    test("CakeContainer Test", () => {
        debug(component);
        expect(component).toMatchSnapshot()
    })

    test('should render with correct number of cakes', () => {
        const wrapper = findByClassName(component, 'default-cake-count');
        expect(wrapper.length).toBe(1)
        const expectedText = "Number of Cakes 10";
        const receivedText = wrapper.text().trim();
        expect(expectedText).toEqual(receivedText)
    });
    test('should call buyCake prop function when the Buy Cake button is clicked', () => {
        const buyCake = findByTestAttr(component, 'test-buy-cake')
        expect(buyCake.exists()).toBe(true);
        buyCake.simulate('click');
        expect(component.find('h2').text()).toContain(`Number of Cakes ${9}`);
        });

})