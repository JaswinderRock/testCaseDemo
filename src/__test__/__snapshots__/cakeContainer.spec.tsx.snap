// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CakeContainer  CakeContainer Test 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CakeContainer)>
    <CakeContainer
      buyCake={[Function]}
      numOfCakes={10}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          <h2
            className="default-cake-count"
            data-test="default-cake-count"
          >
            Number of Cakes 
            10
          </h2>
          <button
            data-test="test-buy-cake"
            onClick={[Function]}
            style={
              Object {
                "background": "blue",
                "borderRadius": "4px",
                "color": "white",
                "display": "flex",
                "fontSize": "24px",
                "margin": "auto",
                "padding": "10px",
                "textAlign": "center",
              }
            }
          >
            Buy Cake
          </button>
        </div>
      </div>
    </CakeContainer>
  </Connect(CakeContainer)>
</Provider>
`;
